extensions/roulette/database/redis_db.py

```diff
@@ -0,0 +1,95 @@
+import logging
+
+from Redis.redis_config import get_redis
+from ..config import config
+from .action import fetch_timeout_roles
+
+from datetime import datetime, timedelta
+from discord import Member, Message
+from typing import Set
+
+_redis_client = get_redis()
+
+logger = logging.getLogger("roulette.database")
+
+
+# Mute functions
+# TODO: Check for successful update
+async def _upload_timeout(duration: timedelta,
+                          message: Message,
+                          target: Member) -> int:
+
+    key = config.redis_key_const() + str(message.guild.id)
+
+    time_unmute = datetime.now() + duration
+    posix_time_unmute = int(time_unmute.timestamp())
+
+    resp = _redis_client.zadd(key, {target.id: posix_time_unmute})
+    return resp
+
+
+async def _apply_timeout_roles(target, duration_label):
+    timeout_ids = config.timeout_roles()
+
+    for timeout_id in timeout_ids:
+        role = target.guild.get_role(int(timeout_id))
+
+        if role:
+            await target.add_roles(role, reason=f"Adding role for {duration_label} timeout")
+
+
+async def add_timeout(target, duration_label, duration, message):
+    await _upload_timeout(duration, message, target)
```

Comment:
imo there's no particular need to have a wrapper here; can just make `_upload_timeout` called `add_timeout` directly.